src/hacl/sha3.rs

```diff
@@ -94,6 +94,89 @@ pub(crate) fn shake256<const BYTES: usize>(data: &[u8]) -> [u8; BYTES] {
     out
 }
 
+pub mod incremental {
```

Comment:
This module needs comments.
Can you file a follow up issue to not make this public but wrap it in a proper API in `digest`?